### ISSUE TITLE
[Jellyfin 10.9.0] Plugin stops working after upgrade to JellyFin 10.9.0.0

### ISSUE THREAD

**Issue by markcs** [Kevinjil/Jellyfin.Xtream#95]
After upgrading to 10.9.0.0, the plugin stops working.

When going into the configuration options of the plugin and saving the configuration, the error message below is seen in the JellyFin server logs and the plugin never works

```
[10:37:23] [ERR] [10] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL POST /Plugins/5d774c35-8567-46d3-a950-9bb8227a0c5d/Configuration.
System.ArgumentException: Refresh task not found
   at Jellyfin.Xtream.Service.TaskService.CancelIfRunningAndQueue(String assembly, String fullName)
   at Jellyfin.Xtream.Plugin.UpdateConfiguration(BasePluginConfiguration configuration)
   at Jellyfin.Api.Controllers.PluginsController.UpdatePluginConfiguration(Guid pluginId)
   at lambda_method858(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

**Comment by seemebreakthis** [comment]
Came to say this.  I just upgraded to 10.9.1 (from 10.8.x).

But not yet 100% sure if it is version related.  And I don't see the same log error as the one above.  Will spend more time to see where the problem is.

Anyone else having problems?

**Comment by markcs** [comment]
@seemebreakthis What errors are you seeing?  The above error is seen when saving the configuration or installing and configuring the plugin on 10.9.0.  It likely has another error if already installed during upgrade and then trying to use it (which also didn't work for me)

**Comment by Kevinjil** [comment]
Jellyfin 1.9.0 has a lot of breaking changes. The plugin has not yet been tested and/or modified for it.

**Comment by seemebreakthis** [comment]
> Jellyfin 1.9.0 has a lot of breaking changes. The plugin has not yet been tested and/or modified for it.

Hope a new version will be ready for 1.9.x soon (along with the much requested feature to organize and sort the channels, yes I know I am being greedy....)

**Comment by seemebreakthis** [comment]
> @seemebreakthis What errors are you seeing? The above error is seen when saving the configuration or installing and configuring the plugin on 10.9.0. It likely has another error if already installed during upgrade and then trying to use it (which also didn't work for me)

Yes mine stopped working and was already installed before upgrade.

**Comment by Kevinjil** [comment]
Fixed by #96
